### PR TITLE
PR11 - Remove AlicaEngine::this from RuntimePlanFactory 

### PR DESCRIPTION
**Pull request by triccyx**
PR11 - Remove AlicaEngine::this from RuntimePlanFactory

### CODE REVIEW

File: alica_engine/src/engine/AlicaContext.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ void AlicaContext::stepEngine()
         if (std::chrono::system_clock::now() > start + timeout) {
             throw std::runtime_error("Got stuck trying to step engine");
         }
-    } while (!_engine->getPlanBase().isWaiting());
+    } while (!_engine->editPlanBase().isWaiting()); //<<<-----edit instedad of get
```
Comment (christianhelm) [inline alica_engine/src/engine/AlicaContext.cpp]: This comment can be removed

File: alica_engine/src/engine/PlanBase.cpp [hunk 2]
```diff
@@ -275,7 +275,8 @@ void PlanBase::run(const Plan* masterPlan)
         if (checkFp) {
             std::queue<RunningPlan*> nextFpEvents;
             {
-                // move fath path events to a local variable. Prevents calling visit() on a RunningPlan which can add a fast path event double locking _loMutex
+                // move fath path events to a local variable. Prevents calling visit() on a RunningPlan which can add a fast path event double locking
```
Comment (christianhelm) [inline alica_engine/src/engine/PlanBase.cpp]: fath -> fast

File: alica_engine/src/engine/PlanBase.cpp [hunk 3]
```diff
@@ -261,7 +261,7 @@ void PlanBase::run(const Plan* masterPlan)
 
         log.iterationEnds(_rootNode);
 
-        //_ae->iterationComplete(); TODO modify when AlicaEngine::iterationComplete will be written
+        //_ae->iterationComplete();
```
Comment (christianhelm) [inline alica_engine/src/engine/PlanBase.cpp]: Why is the instruction still commented but the TODO removed?

Comment (triccyx) [reply alica_engine/src/engine/PlanBase.cpp]: ok

**Review by christianhelm — APPROVED** [review]

File: alica_engine/include/engine/PlanBase.h [hunk 4]
```diff
@@ -83,16 +83,16 @@ class PlanBase
     Logger& _logger;
     const IAlicaCommunication& _communicator;
     IRoleAssignment& _roleAssignment;
-    SyncModule& _syncModule;
+    SyncModule& _synchModule;
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/PlanBase.h]: "sync" is more standard as an abbreviation

File: alica_engine/include/engine/PlanBase.h [hunk 5]
```diff
@@ -83,16 +83,16 @@ class PlanBase
...
     AuthorityManager& _authorityManager;
     TeamObserver& _teamObserver;
     TeamManager& _teamManager;
     const PlanRepository& _planRepository;
-    bool _stepEngine;
-    bool _stepCalled;
+    bool& _stepEngine;
+    bool& _stepCalled;
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/PlanBase.h:92]: This is weird.  Why do we need these to be mirrored from the engine?

File: alica_engine/include/engine/RuntimePlanFactory.h [hunk 6]
```diff
@@ -11,15 +11,19 @@ class Plan;
 class BasicPlan;
 class IPlanCreator;
 class IAlicaWorldModel;
+class IAlicaTraceFactory;
+class TeamManager;
+class IAlicaTimerFactory;
 
 /**
- * Construct a runtime BasicPlan instance based
+ * @brief
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/RuntimePlanFactory.h]: Empty brief?

File: alica_engine/include/engine/util/idFunctions.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/util/idFunctions.h]: remove

File: alica_engine/include/engine/util/idFunctions.h [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ * This method can be used, e.g., for passing a part of a ROS
+ * message and receiving a pointer to a corresponding Identifier object.
+ */
+
+/**
+ * Generates random ID.
+ * @return The ID
+ */
+
+inline alica::AgentId GenerateID()
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/util/idFunctions.h]: This function doesn't appear to be used, nor is the description accurate.  Can this file be removed?

The formatting/name of this file also don't follow convention

File: alica_engine/src/engine/PlanBase.cpp [hunk 9]
```diff
@@ -375,15 +376,15 @@ RunningPlan* PlanBase::makeRunningPlan(const Plan* plan, const Configuration* co
...
 RunningPlan* PlanBase::makeRunningPlan(const Behaviour* behaviour, const Configuration* configuration)
 {
-    _runningPlans.emplace_back(new RunningPlan(_configChangeListener, _clock, _worldModel, nullptr, _teamObserver, _teamManager, _planRepository,
+    _runningPlans.emplace_back(new RunningPlan(_configChangeListener, _clock, _worldModel, _runTimePlanFactory, _teamObserver, _teamManager, _planRepository,
             _runTimeBehaviourFactory, _resultStore, _solvers, behaviour, configuration));
     return _runningPlans.back().get();
 }
 
-RunningPlan* PlanBase::makeRunningPlan(const PlanType* planType, const Configuration* configuration)
+RunningPlan* PlanBase::makeRunningPlan(const PlanType* pt, const Configuration* configuration)
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/PlanBase.cpp]: Why?

File: alica_tests/src/test/test_task_assignment.cpp [hunk 10]
```diff
@@ -145,6 +145,7 @@ TEST_F(TaskAssignmentTest, switchEntryPoints)
             ae->getPlanRepository().getPlans().find(1407152758497), nullptr);
 
     // fake inform the team observer about roles of none existing robots
+    // alica::RunningPlan* rp = new RunningPlan(ae, ae->getPlanRepository().getPlans().find(1407152758497), nullptr);
```
Comment (Dale-Koenig) [inline alica_tests/src/test/test_task_assignment.cpp]: Please clean up this PR and make sure all changes are deliberate/meaningful.  In general please try to keep the "most immediate" PR that is currently under review to be clean of unintentional/irrelevant changes.

File: alica_engine/include/engine/PlanBase.h (hunk 4 above)
Comment (bjoernschroeder) [inline alica_engine/include/engine/PlanBase.h]: Is there any reason to rename it to _synchModule?

File: alica_engine/include/engine/RuntimePlanFactory.h [hunk 11]
```diff
@@ -11,15 +11,19 @@ class Plan;
 class BasicPlan;
```
Comment (bjoernschroeder) [inline alica_engine/include/engine/RuntimePlanFactory.h:10]: forward declaration of engine can be removed

File: alica_engine/src/engine/PlanBase.cpp [hunk 12]
```diff
@@ -29,22 +29,22 @@ namespace alica
...
+        const PlanRepository& planRepository, bool& stepEngine, bool& stepCalled, IAlicaWorldModel* worldModel, const RuntimePlanFactory& runTimePlanFactory,
+        const RuntimeBehaviourFactory& runTimeBehaviourFactory, VariableSyncModule& resultStore,
         const std::unordered_map<size_t, std::unique_ptr<ISolverBase>>& solvers)
         : _configChangeListener(configChangeListener)
         , _clock(clock)
         , _logger(log)
         , _communicator(communicator)
         , _roleAssignment(roleAssignment)
-        , _syncModule(syncModule)
+        , _synchModule(synchModule)
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/PlanBase.cpp]: _syncModule

File: alica_engine/src/engine/RuntimePlanFactory.cpp [hunk 13]
```diff
@@ -10,8 +10,11 @@
 namespace alica
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/RuntimePlanFactory.cpp:9]: Remove engine include

File: alica_engine/include/engine/PlanBase.h (hunk 4 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanBase.h]: ok

File: alica_engine/src/engine/PlanBase.cpp (hunk 12 above)
Comment (triccyx) [reply alica_engine/src/engine/PlanBase.cpp]: ok

File: alica_engine/src/engine/RuntimePlanFactory.cpp (hunk 13 above)
Comment (triccyx) [reply alica_engine/src/engine/RuntimePlanFactory.cpp:9]: PlanBase uses 
`AlicaEngine::abort`

File: alica_engine/include/engine/RuntimePlanFactory.h (hunk 11 above)
Comment (triccyx) [reply alica_engine/include/engine/RuntimePlanFactory.h:10]: ok

File: alica_engine/include/engine/PlanBase.h (hunk 4 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanBase.h]: ok

File: alica_engine/include/engine/PlanBase.h (hunk 5 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanBase.h:92]: PR16 corrects this problem

File: alica_engine/include/engine/RuntimePlanFactory.h (hunk 6 above)
Comment (triccyx) [reply alica_engine/include/engine/RuntimePlanFactory.h]: ok

File: alica_engine/include/engine/util/idFunctions.h (hunk 7 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: ok

File: alica_engine/include/engine/util/idFunctions.h (hunk 8 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: This function replaces the one in  TeamManager
But there is a merge problem here

File: alica_engine/include/engine/PlanBase.h [hunk 14]
```diff
@@ -44,9 +44,9 @@ class PlanBase
 {
 public:
     PlanBase(ConfigChangeListener& configChangeListener, const AlicaClock& clock, Logger& log, const IAlicaCommunication& communicator,
-            IRoleAssignment& roleAssignment, SyncModule& syncModule, AuthorityManager& authorityManager, TeamObserver& teamObserver, TeamManager& teamManager,
-            const PlanRepository& planRepository, bool stepEngine, IAlicaWorldModel* worldModel, const std::unique_ptr<RuntimePlanFactory>& runTimePlanFactory,
-            const std::unique_ptr<RuntimeBehaviourFactory>& runTimeBehaviourFactory, VariableSyncModule& resultStore,
+            IRoleAssignment& roleAssignment, SyncModule& synchModule, AuthorityManager& authorityManager, TeamObserver& teamObserver, TeamManager& teamManager,
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/PlanBase.h]: synch->sync everywhere remaining

File: alica_engine/include/engine/PlanBase.h [hunk 15]
```diff
@@ -88,11 +88,11 @@ class PlanBase
     TeamObserver& _teamObserver;
     TeamManager& _teamManager;
     const PlanRepository& _planRepository;
-    bool _stepEngine;
-    bool _stepCalled;
+    bool& _stepEngine;
+    bool& _stepCalled;
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/PlanBase.h:92]: I don't see anything in PlanBase.cpp using these, so I think you can take them out

File: alica_engine/include/engine/RunningPlan.h [hunk 16]
```diff
@@ -234,12 +231,16 @@ class RunningPlan : public std::enable_shared_from_this<RunningPlan>
     const AlicaClock& getAlicaClock() const { return _clock; };
     VariableSyncModule& editResultStore() const { return _resultStore; }
 
+    //[[deprecated("temporary method")]]
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/RunningPlan.h:234]: temporary until when?

File: alica_engine/include/engine/util/idFunctions.h (hunk 7 above)
Comment (Dale-Koenig) [reply alica_engine/include/engine/util/idFunctions.h]: Why did you you resolve this?

File: alica_engine/include/engine/util/idFunctions.h [hunk 17]
```diff
@@ -0,0 +1,28 @@
+
+#pragma once
+
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/util/idFunctions.h]: Rename file to IdFunctions.h, or consider combining with another util file

File: alica_engine/include/engine/util/idFunctions.h (hunk 7 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: done

File: alica_engine/include/engine/util/idFunctions.h (hunk 17 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: done

File: alica_engine/include/engine/RunningPlan.h (hunk 16 above)
Comment (triccyx) [reply alica_engine/include/engine/RunningPlan.h:234]: In PR16 have been removed

File: alica_engine/include/engine/PlanBase.h (hunk 15 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanBase.h:92]: No there are references in PlanBase.cpp line 143,341 ecc...

File: alica_engine/include/engine/PlanBase.h (hunk 14 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanBase.h]: Ok

File: alica_tests/src/test/test_task_assignment.cpp (hunk 10 above)
Comment (triccyx) [reply alica_tests/src/test/test_task_assignment.cpp]: ok

File: alica_engine/src/engine/PlanBase.cpp (hunk 9 above)
Comment (triccyx) [reply alica_engine/src/engine/PlanBase.cpp]: ok, merge problem already changed in the past

**Review by Dale-Koenig — APPROVED** [review]

**Pull request merged**